types/core.tst.ts

```diff
@@ -0,0 +1,260 @@
+import { expect, test, describe } from 'tstyche'
+import {
+  encodeCursor,
+  decodeCursor,
+  PAGINATION_MODE,
+  connectionDirective
+} from './core.js'
+import type { ConnectionArgs, ConnectionResolverResponse } from './core.js'
```

Comment:
Since this PR is an experiment, would you mind adding the following to `package.json`:

```json
"exports": {
  ".": "./index.js"
},
```

After that you can rewrite these imports: `import xyz from 'relay-pagination-directive'`. This way the root `index.d.ts` will be used an issues with it will be revealed. For instance, it should not have extension less imports.

---

The change in `package.json` allows using self-reference import. See Node.js documentation for more details: https://nodejs.org/docs/latest/api/packages.html#self-referencing-a-package-using-its-name

By the way, you can use `import xyz from 'relay-pagination-directive'` in unit tests as well.